### PR TITLE
Upgrade to tensorflow 2.3.1

### PR DESCRIPTION
**Pull request by frankfliu**
## Channel for questions ##
Before or while filing an issue please feel free to join our [Slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/zt-ar91gjkz-qbXhr1l~LFGEIEeGBibT7w) to get in touch with development team, ask questions, find out what's cooking and more!

## Description ##
(Brief description of what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- [ ] Code is well-documented: 
    - For user-facing API changes, Java doc has been updated. 
    - For new examples, README.md is added to explain the what the example does.
- [ ] To the my best knowledge, [examples](https://github.com/awslabs/djl/tree/master/examples) and [jupyter notebooks](https://github.com/awslabs/djl/tree/master/jupyter) are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, Java doc)
- [ ] Feature2, tests, (and when applicable, Java doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: tensorflow/tensorflow-model-zoo/build.gradle [hunk 1]
```diff
@@ -13,7 +13,7 @@ dependencies {
     testImplementation("org.testng:testng:${testng_version}") {
         exclude group: "junit", module: "junit"
     }
-    testRuntimeOnly "ai.djl.tensorflow:tensorflow-native-auto:${tensorflow_version}"
+    testRuntimeOnly "ai.djl.tensorflow:tensorflow-native-auto:${tensorflow_version}-SNAPSHOT"
```
Comment (lanking520) [inline tensorflow/tensorflow-model-zoo/build.gradle:16]: If the public version pushlished, do we still need to keep SNAPSHOT?

**Review by zachgk — APPROVED** [review]

**Review by lanking520 — APPROVED** [review]

File: tensorflow/tensorflow-model-zoo/build.gradle (hunk 1 above)
Comment (frankfliu) [reply tensorflow/tensorflow-model-zoo/build.gradle:16]: We can change it back once we released 2.3.1 to maven
